### PR TITLE
Extension resturcture

### LINKED ISSUE #450: Add directory structures to extensions
Similar to how the the catalog, collection, and item specs have examples and json-schema directories, the extensions directories should follow the same convention.

Relates to #445

### PR DESCRIPTION
**Pull request by jbants**
This PR places extension example and schema files in separate directories. This is done to align with the catalog, collection and item spec directory structures.

Closes #450

### CODE REVIEW

File: extensions/README.md [hunk 1]
```diff
@@ -127,3 +127,12 @@ An example of this can be seen in a Landsat example:
     "l8:image_quality_oli": 9
   },
 ```
+
+### Directory Structure
+
+A STAC extension can have references to additional schemas within the extension schema. An example of this is the STAC item spec and the accompyning geojson spec.
```
Comment (m-mohr) [inline extensions/README.md]: accompyning is a typo

File: extensions/README.md [hunk 2]
```diff
@@ -127,3 +127,12 @@ An example of this can be seen in a Landsat example:
...
+
+In order to keep these files together to preserve relative `ref` links the files should be grouped.
```
Comment (m-mohr) [inline extensions/README.md]: Is the `ref` meant to be `href`? And the sentence sounds a little strange overall. Is there somewhere a word missing?

Comment (jbants) [reply extensions/README.md]: I've changed it to `$ref`as it's used in the json schema files and cleaned up the description a bit.

**Comment by jbants** [comment]
Thanks @m-mohr!

**Pull request merged**